twake/backend/node/src/core/platform/services/auth/web/jwt.ts

```diff
@@ -3,14 +3,32 @@ import fastifyJwt from "fastify-jwt";
 import fp from "fastify-plugin";
 import config from "../../../../config";
 
+type JwtType = {
+  sub: string;
+  exp: number;
+  refresh_exp: number;
+  updated_at: number;
+  org: {
+    [wompanyId: string]: {
+      role: string; //Not implemented
+      wks: {
+        [workspaceId: string]: {
+          adm: boolean;
+        };
+      };
+    };
+  };
+};
+
 const jwtPlugin: FastifyPluginCallback = (fastify, _opts, next) => {
   fastify.register(fastifyJwt, {
     secret: config.get("auth.jwt.secret"),
   });
 
   fastify.decorate("authenticate", async (request: FastifyRequest) => {
     try {
-      request.currentUser = await request.jwtVerify();
+      const jwt: JwtType = await request.jwtVerify();
+      request.currentUser = { id: jwt.sub };
```

Comment:
I suggest to keep the decoded jwt in the `currentUser` like:

```js
request.currentUser = {id: jwt.sub, jwt}
```

Reply:
I'll let you do it because it'll modify the User type and then the WebSocketUser type. I don't want the WebSocketUser to contain all this data.

Reply:
OK